internal/evaluator/conftest_evaluator.go

```diff
@@ -427,6 +418,20 @@ func addMetadataToResults(r *output.Result, rule rule.Info) {
 	if len(rule.Collections) > 0 {
 		r.Metadata[metadataCollections] = rule.Collections
 	}
+
+	// If the rule has been effective for a long time, we'll consider
+	// the effective_on date not relevant and not bother including it
+	if effectiveOnString, ok := r.Metadata[metadataEffectiveOn]; ok {
```

Comment:
This makes it a bit safer, `ok` now tells if the key exists and if the value is a string, which makes the cast below unnecessary

```suggestion
	if effectiveOnString, ok := r.Metadata[metadataEffectiveOn].(string); ok {
```